### PR TITLE
circle ci: save geth logs

### PR DESCRIPTION
**Pull request by hackaugusto**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -258,6 +258,7 @@ jobs:
     environment:
       RAIDEN_TESTS_SYNAPSE_BASE_DIR: /home/circleci/.cache/synapse
       RAIDEN_TESTS_SYNAPSE_LOGS_DIR: /tmp/synapse-logs
+      RAIDEN_TESTS_GET_DATADIR: /tmp/geth
```
Comment (ulope) [inline .circleci/config.yml]: `GET` -> `GETH`

File: raiden/tests/integration/fixtures/blockchain.py [hunk 2]
```diff
@@ -13,6 +15,8 @@
 
 # pylint: disable=redefined-outer-name,too-many-arguments,unused-argument,too-many-locals
 
+_GETH_DATADIR = os.environ.get('RAIDEN_TESTS_GET_DATADIR', False)
```
Comment (ulope) [inline raiden/tests/integration/fixtures/blockchain.py]: Same as above

**Comment by hackaugusto** [comment]
> Would it make more sense to only set the log dir via the env variable to avoid having multiple runs of the geth fixture re-use the same datadir?

I don't follow. Local runs will use the `tmpdir` fixture, which is different for each test function, circle builds will use the `/tmp/geth` directory, in order to use it with the artifacts directive.

**Comment by ulope** [comment]
> circle builds will use the /tmp/geth directory, in order to use it with the artifacts directive.

Yes that's what I meant. Since the fixture is function scoped it is run again for every test function. I was just wondering if reusing the same datadir could cause problems. 

My suggestion was to only move the log dir to a well known location on ci and keep using the tmpdir fixture for the datadir.

**Comment by hackaugusto** [comment]
> I was just wondering if reusing the same datadir could cause problems.

The datadir is not shared, only the logdir (I don't know, however, if the logs are trimmed on each run).

I changed the `blockchain_key_seed` fixture to guarantee that we will have a different privatekey for each test run:

https://github.com/raiden-network/raiden/blob/5530b9a2f99152b1d336fa198390a1c3a741470a/raiden/tests/fixtures/variables.py#L245-L253

and the datadir is computed based on the first few bytes of the private key:

https://github.com/raiden-network/raiden/blob/5530b9a2f99152b1d336fa198390a1c3a741470a/raiden/tests/utils/geth.py#L265-L272

**Pull request merged**